Sources/PackageDescription/PackageDescriptionSerialization.swift

```diff
@@ -224,18 +224,65 @@ extension SystemPackageProvider: Encodable {
 
 extension Target.PluginCapability: Encodable {
     private enum CodingKeys: CodingKey {
-        case type
+        case type, intent, permissions
     }
 
     private enum Capability: String, Encodable {
-        case buildTool
+        case buildTool, command
     }
 
     public func encode(to encoder: Encoder) throws {
         var container = encoder.container(keyedBy: CodingKeys.self)
         switch self {
         case ._buildTool:
             try container.encode(Capability.buildTool, forKey: .type)
+        case ._command(let intent, let permissions):
+            try container.encode(Capability.command, forKey: .type)
+            try container.encode(intent, forKey: .intent)
+            try container.encode(permissions, forKey: .permissions)
+        }
+    }
+}
+
+extension PluginCommandIntent: Encodable {
+    private enum CodingKeys: CodingKey {
+        case type, verb, description
+    }
+
+    private enum IntentType: String, Encodable {
+        case documentationGeneration, sourceCodeFormatting, custom
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        switch self {
+        case .documentationGeneration:
+            try container.encode(IntentType.documentationGeneration, forKey: .type)
+        case .sourceCodeFormatting:
+            try container.encode(IntentType.sourceCodeFormatting, forKey: .type)
+        case .custom(let verb, let description):
+            try container.encode(IntentType.custom, forKey: .type)
+            try container.encode(verb, forKey: .verb)
+            try container.encode(description, forKey: .description)
+        }
+    }
+}
+
+extension PluginPermission: Encodable {
+    private enum CodingKeys: CodingKey {
+        case type, reason
+    }
+
+    private enum PermissionType: String, Encodable {
+        case packageWritability
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        switch self {
+        case .packageWritability(let reason):
+            try container.encode(PermissionType.packageWritability, forKey: .type)
+            try container.encode(reason, forKey: .reason)
         }
```

Comment:
enums coddle conformance is automatically synthesized now, so maybe remove all this custom encoding?

Reply:
I added this a while ago and I think there was some reason why it had to be there — I will check again.  It's a very good point.

Reply:
Oh I know what it was.  At the time we still wanted to be able to build with Swift 5.4 (which is still our tools requirement), and the synthesis wasn't added until 5.5 AFAIK.

Also even with 5.5 it does mean that we need to add the `Encodable` conformance in the PackageDescription API in order to get `Codable` synthesis (can't be in an extension) but I don't think that's a major API change that would break anyone.

I think in another PR comment you said there's nothing preventing SwiftPM from using 5.5 tools version, so maybe we should do that before merging this and then remove this?

Reply:
The other thing is that then we'd need to make sure that the code that is reading JSON on the libSwiftPM side agrees on the keys in the JSON, though we can always adjust it to match the synthesized ones.  IIRC there's a special `JSON` type there that predates `Codable`.